src/content/docs/apm/agents/python-agent/python-agent-api/setllmtokencountcallback-python-agent-api.mdx

```diff
@@ -0,0 +1,160 @@
+---
+title: set_llm_token_count_callback (Python agent API)
+type: apiDoc
+shortDescription: Registers a callback to calculate LLM token counts
+tags:
+  - Agents
+  - Python agent
+  - Python agent API
+metaDescription: 'Python API: This call registers a callback to calculate LLM token counts'
+redirects:
+  - /docs/agents/python-agent/python-agent-api/setllmtokencountcallback-python-agent-api
+  - /docs/agents/python-agent/python-agent-api/set_llm_token_count_callback
+freshnessValidatedDate: never
+---
+
+## Syntax
+
+```py
+newrelic.agent.set_llm_token_count_callback(callback, application=None)
+```
+
+Registers a callback function which will be used for calculating token counts on Llm events.
+
+
+## Requirements
+
+Python agent version 9.8.0 or higher.
+
+
+## Description
+
+This API registers a callback to calculate and store token counts on `LlmEmbedding` and `LlmChatCompletionMessage` events.
+This function will typically only be used if `ai_monitoring.record_content.enabled` is False as this prevents the agent from sending AI content to New Relic.
+If you would still like to capture token counts for Llm events in this instance, you can implement a callback in your app code to determine the token counts locally and send this information to New Relic.
+In most cases, this API will be called exactly once soon after the agent has initialized. However, you may make multiple calls to this API if you would like.
```

Comment:
This isn't a pre-requisite in Python's case. They can just put this call at the top of their main file.
```suggestion
In most cases, this API will be called exactly once. However, you may make multiple calls to this API if you would like.
```